src/Model/Attach.php

```diff
@@ -328,4 +330,43 @@ public static function delete(array $conditions, array $options = []): bool
 
 		return DBA::delete('attach', $conditions, $options);
 	}
+
+	public static function setPermissionFromBody(array $post)
+	{
+		if (preg_match_all("/\[attachment\](.*?)\[\/attachment\]/ism", $post['body'], $matches, PREG_SET_ORDER)) {
+			foreach ($matches as $attachment) {
+				if (Network::isLocalLink($attachment[1]) && preg_match('|.*?/attach/(\d+)|', $attachment[1], $match)) {
+					$fields = [
+						'allow_cid' => $post['allow_cid'], 'allow_gid' => $post['allow_gid'],
+						'deny_cid' => $post['deny_cid'], 'deny_gid' => $post['deny_gid']
+					];
+					self::update($fields, ['id' => $match[1], 'uid' => $post['uid']]);
+				}
+			}
+		}
```

Comment:
`$matches` will be an empty array if there aren't any matches, removing the need for the `if`.
```suggestion
		preg_match_all("/\[attachment\](.*?)\[\/attachment\]/ism", $post['body'], $matches, PREG_SET_ORDER);
		foreach ($matches as $attachment) {
			if (Network::isLocalLink($attachment[1]) && preg_match('|.*?/attach/(\d+)|', $attachment[1], $match)) {
				$fields = [
					'allow_cid' => $post['allow_cid'], 'allow_gid' => $post['allow_gid'],
					'deny_cid' => $post['deny_cid'], 'deny_gid' => $post['deny_gid']
				];
				self::update($fields, ['id' => $match[1], 'uid' => $post['uid']]);
			}
		}
```

Reply:
Done